src/lib/ndpi_utils.c

```diff
@@ -1416,30 +1415,32 @@ int ndpi_flow2json(struct ndpi_detection_module_struct *ndpi_struct,
     ndpi_patchIPv6Address(src_name), ndpi_patchIPv6Address(dst_name);
   }
 
-  ndpi_serialize_string_string(serializer, "src_ip", src_name);
-  ndpi_serialize_string_string(serializer, "dest_ip", dst_name);
-  if(src_port) ndpi_serialize_string_uint32(serializer, "src_port", src_port);
-  if(dst_port) ndpi_serialize_string_uint32(serializer, "dst_port", dst_port);
+  ndpi_serialize_string_string(&serializer, "src_ip", src_name);
+  ndpi_serialize_string_string(&serializer, "dest_ip", dst_name);
+  if(src_port) ndpi_serialize_string_uint32(&serializer, "src_port", src_port);
+  if(dst_port) ndpi_serialize_string_uint32(&serializer, "dst_port", dst_port);
 
   switch(l4_protocol) {
   case IPPROTO_TCP:
-    ndpi_serialize_string_string(serializer, "proto", "TCP");
+    ndpi_serialize_string_string(&serializer, "proto", "TCP");
     break;
 
   case IPPROTO_UDP:
-    ndpi_serialize_string_string(serializer, "proto", "UDP");
+    ndpi_serialize_string_string(&serializer, "proto", "UDP");
     break;
 
   case IPPROTO_ICMP:
-    ndpi_serialize_string_string(serializer, "proto", "ICMP");
+    ndpi_serialize_string_string(&serializer, "proto", "ICMP");
     break;
 
   default:
-    ndpi_serialize_string_uint32(serializer, "proto", l4_protocol);
+    ndpi_serialize_string_uint32(&serializer, "proto", l4_protocol);
     break;
   }
 
-  return(ndpi_dpi2json(ndpi_struct, flow, l7_protocol, serializer));
+  int res = ndpi_dpi2json(ndpi_struct, flow, l7_protocol, &serializer);
+  ndpi_term_serializer(&serializer);
```

Comment:
After calling that function, the resulting JSON buffer won't be accessible anymore.